File: src/nahyeon/TwoPointers/boj_3273_두수의합.java
package nahyeon.TwoPointers;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class boj_3273_두수의합 {

    static FastReader scan = new FastReader();

    static int N, X, A[];

    static void input() {
        N = scan.nextInt();
        A = new int[N + 1];
        for (int i = 1; i <= N; i++) {
            A[i] = scan.nextInt();
        }
        X = scan.nextInt();
    }

    static void pro() {
        Arrays.sort(A, 1, N + 1);
        int L = 1, R = N, ans = 0;
        while (L < R) {
            int sum = A[L] + A[R];
            if (sum == X) ans++;

            if (sum >= X) R--;
            else L++;
        }
        System.out.println(ans);
    }

    public static void main(String[] args) {
        input();
        pro();
    }

    static class FastReader {

        BufferedReader br;
        StringTokenizer st;

        public FastReader() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        String next() {
            while (st == null || !st.hasMoreTokens()) {
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() {
            return Integer.parseInt(next());
        }
    }
}

File: src/hyewon/TwoPointers/boj_3273_두 수의 합.kt
package hyewon.TwoPointers

import java.io.*
import java.util.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val num = br.readLine().toInt()
    val arr = IntArray(num)
    val line = StringTokenizer(br.readLine())
    for(i in 0 until num) {
        arr[i] = line.nextToken().toInt()
    }
    arr.sort()
    var (sum, cnt, start, end) = arrayOf(br.readLine().toInt(), 0, 0, num -1)
    var tmp : Int

    while (start < end) {
        tmp = arr[start] + arr[end]
        if (tmp == sum) {
            cnt++
        }
        if (tmp <= sum) {
            start++
        } else {
            end--
        }
    }

    bw.write("$cnt")
    bw.flush()
    bw.close()
    br.close()
}